purity/gc/gc.go

```diff
@@ -308,24 +289,50 @@ func (c *collector) markReferencedBlobs(ar *ppb.ActionResult) error {
 	// Mark all the inputs as well. There are some fringe cases that make things awkward
 	// and it means things look more sensible in the browser.
 	inputSize, dirs, _ := c.inputDirs(dg)
-	c.mutex.Lock()
-	defer c.mutex.Unlock()
-	c.referencedBlobs[ar.Hash] = struct{}{}
-	for _, f := range result.OutputFiles {
-		c.referencedBlobs[f.Digest.Hash] = struct{}{}
-		size += f.Digest.SizeBytes
-	}
```

Comment:
This part is done before `inputDirs` with your changes. Is this OK? i.e. is the order important here or not?

Reply:
I can't see that ordering is important here. It looks like it was originally done here so c.referencedBlobs was covered by the lock. Now that's been moved to it's own function it shouldn't be an issue